Project/dotnet-core/ConsoleApp/Program.cs

```diff
@@ -0,0 +1,217 @@
+                    Int64 Position = fileStream.Seek(desiredSeekLocation, SeekOrigin.Begin); //Position the file
+                    contentLength = fileStream.Length;
+                    MI.Open_Buffer_Init(contentLength, Position); //Informing MediaInfo we have seek
+                }
+            }
+            while (numberOfBytesRead > 0);
+
+            Console.WriteLine($"Total Bytes read: {totalNumberOfBytesRead}/{contentLength} = {totalNumberOfBytesRead / (float)contentLength}");
+
+            // This is the end of the stream, MediaInfo must finish some work
+            MI.Open_Buffer_Finalize();
+
+            // Use MediaInfoLib as needed
+            MI.Option("Complete", "1");
+            string ToDisplay = MI.Inform();
+
+            return ToDisplay;
+        }
+
+
+
+        //+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+        //
+        // This example demonstrates how to parse a file in Azure Blob Storage using a connection string for auth.
+        // MediaInfo does not need to parse the whole remote file.
+        // 
+        // Uses BlobBaseClient.DownloadAsync for byte-range downloads.
+        // Ref: https://docs.microsoft.com/en-us/dotnet/api/azure.storage.blobs.specialized.blobbaseclient.downloadasync
+        // Ref: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs#L243
+        // 
+        // Add the following nuget PackageReference your project file:
+        //   Azure.Storage.Blobs (https://www.nuget.org/packages/Azure.Storage.Blobs/)
+        // 
+        // Add to your implementation class:
+        //   using Azure.Storage.Blobs.Specialized;
+        //
+        //+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+        static async Task<string> ExampleReadingFromAzureBlobStorage()
+        {
+            // Set the following for your content:
+            var fileUri = new Uri("https://mystorageaccount.blob.core.windows.net/blendercontainer/bbb_sunflower_1080p_30fps_normal.mp4");
+            var connectionString = "DefaultEndpointsProtocol=https;AccountName=mystorageaccount;AccountKey=...";
+
+            var blobUriBuilder = new BlobUriBuilder(fileUri); // The blobUriBuilder simplifies the parsing of Azure Storage Uris
+            BlobBaseClient blobBaseClient = new BlobBaseClient(connectionString, blobUriBuilder.BlobContainerName, blobUriBuilder.BlobName);
+
+            // Initialize MediaInfo
+            var MI = new MediaInfo();
+
+            // Do a HEAD on the file:
+            var blobProperties = await blobBaseClient.GetPropertiesAsync();
+            long contentLength = blobProperties.Value.ContentLength;
+
+            // Prepare MediaInfo for buffered reading
+            MI.Open_Buffer_Init(contentLength, 0);
+
+            // The parsing loop using a fixed memory buffer for reading and marshaling
+            const int localByteBufferSize = 1 * 1024 * 1024;
```

Comment:
@JeromeMartinez , this buffer seems too large for calls into LibMediaInfo.  The .Inform call will return less data if the buffer size is too large.  First it drops the audio tracks, then in the extreme of a 16MB buffer, it also drops the video track information (all that is left is the container info).
Do you know of some sort of max_size for the buffer passed to Open_Buffer_Continue ?

Reply:
Using and MP4 with an ftyp, free, mdat, moof top-level box layout:

A 4KB buffer will provide the same JSON results as the compiled / released app, whereas a 5KB buffer will have inaccurate video StreamSize and inaccurate audio StreamSize and BitRate (versus the app or 4KB buffer).
The 5KB and 4KB have similar but different seek request patterns, with many matching byte offset requests.  They both read their initial buffers, seek to the moof, (which is 6715 bytes) read it in two sequential reads, then seek back into the mdat and perform lots of reads.
A 6KB (and above) buffer will read the initial buffer, seek to the moof, read it and then we hit EOF.  
I also tried a 6715b (the moof size) and 6714b buffer, to test edge cases around the moof boxe size.  These had the same behavior as 6KB and above.